src/main/java/com/epam/javacourse/hometask4/Greenhouse.java

```diff
@@ -0,0 +1,68 @@
+package com.epam.javacourse.hometask4;
+
+import java.util.Arrays;
+
+public class Greenhouse implements Informational, Watered, Changeable {
+
+    private float temperature;
+    private Plant[] plants;
+
+    public Greenhouse(Plant[] plants) {
+        this.plants = Arrays.copyOf(plants, plants.length);
```

Comment:
You don't need to create a copy of an array to set the property.  this.plant = plants will do the trick